examples/eks_cluster_readonly/main.tf

```diff
@@ -0,0 +1,57 @@
+# Your EKS cluster access configuration.
+
+provider "aws" {
+  region     = var.cluster_region
+  access_key = var.aws_access_key_id
+  secret_key = var.aws_secret_access_key
+}
+
+data "aws_eks_cluster" "cluster" {
+  name = var.cluster_name
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = var.cluster_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.cluster.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority[0].data)
+    token                  = data.aws_eks_cluster_auth.cluster.token
+  }
+}
+
+# Your CAST.AI Eks configuration
+
+provider "castai" {
+  api_token = var.castai_api_token
+}
+
+resource "castai_eks_cluster" "my_castai_cluster" {
+  account_id = var.aws_account_id
+  region     = var.cluster_region
+  cluster    = var.cluster_name
+
+  # providing the credentials will turn on autoscaling mode.
+  # access_key_id = var.aws_access_key_id
+  # secret_access_key = var.aws_secret_access_key
+}
+
+data "castai_agent_settings" "agent_settings" {}
+
+resource "helm_release" "castai_agent" {
+  name            = "castai-agent"
+  repository      = "https://castai.github.io/helm-charts"
+  chart           = "castai-agent"
+  cleanup_on_fail = true
+
+  set {
+    name  = "provider"
+    value = "eks"
+  }
+  set_sensitive {
+    name  = "apiKey"
+    value = data.castai_agent_settings.agent_settings.token
```

Comment:
Shouldn't this be set by calling backend to register new cluster?
What we should set is a token to call backend initially.

Reply:
Currently, api.cast.ai provides the agent token and script before registering any cluster. As agreed, we're moving the token output into `castai_eks_cluster` for convenience and to address ease of supporting cluster-controller in the future.